src/test/scala/cromwell/binding/types/WdlObjectTypeSpec.scala

```diff
@@ -0,0 +1,106 @@
+package cromwell.binding.types
+
+import cromwell.binding.values.{WdlInteger, WdlString, WdlMap, WdlObject}
+import cromwell.parser.WdlParser.SyntaxError
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.util.{Failure, Success}
+
+class WdlObjectTypeSpec extends FlatSpec with Matchers {
+  val abcObject = WdlObject(Map(
+    "a" -> WdlString("one"),
+    "b" -> WdlString("two"),
+    "c" -> WdlString("three")
+  ))
+
+  val coerceableMap = WdlMap(WdlMapType(WdlStringType, WdlStringType), Map(
+    WdlString("a") -> WdlString("one"),
+    WdlString("b") -> WdlString("two"),
+    WdlString("c") -> WdlString("three"))
+  )
+
+  val nonCoerceableMap1 = WdlMap(WdlMapType(WdlIntegerType, WdlStringType), Map(
+    WdlInteger(1) -> WdlString("one"),
+    WdlInteger(2) -> WdlString("two"),
+    WdlInteger(3) -> WdlString("three"))
+  )
+
+  val nonCoerceableMap2 = WdlMap(WdlMapType(WdlStringType, WdlObjectType), Map(
+    WdlString("a") -> WdlObject(Map.empty),
+    WdlString("b") -> WdlObject(Map.empty),
+    WdlString("c") -> WdlObject(Map.empty))
+  )
+  
+  "WdlObject" should "stringify its value" in {
+    abcObject.toWdlString shouldEqual "object {a: \"one\", b: \"two\", c: \"three\"}"
+  }
+
+  it should "stringify its type" in {
+    abcObject.wdlType.toWdlString shouldEqual "Object"
+  }
+
+  it should "convert WDL source code to WdlMap" in {
+    WdlObjectType.fromWdlString("object {a: \"one\", b: \"two\", c: \"three\"}") shouldEqual abcObject
+  }
+
+  it should "coerce a coerceable map into a WdlObject" in {
+    WdlObjectType.coerceRawValue(coerceableMap) match {
+      case Success(v) => 
+        v.wdlType shouldEqual WdlObjectType
+        v.toWdlString shouldEqual abcObject.toWdlString
+      case Failure(f) => fail("Failed to coerce a map to an object")
+    }
+  }
+
+  it should "NOT successfully coerce a NON coerceable map into a WdlObject" in {
+    WdlObjectType.coerceRawValue(nonCoerceableMap1) match {
+      case Success(v) => fail("should not have succeeded")
+      case Failure(f) => // expected
+    }
+  }
+
+  it should "NOT successfully coerce a NON coerceable map into a WdlObject (2)" in {
+    WdlObjectType.coerceRawValue(nonCoerceableMap2) match {
+      case Success(v) => fail("should not have succeeded")
+      case Failure(f) => // expected
+    }
+  }
+
+  it should "NOT successfully convert WDL source code to WdlMap if passed a bogus AST" in {
+    try {
+      WdlObjectType.fromWdlString("workflow wf{}")
```

Comment:
Hmm I guess it could, it's only used in one place though (+ in the tests) to convert a symbol DB entry to a WdlValue, which should never fail.
I think this could be included as part of a larger refactoring/robustifying, I'll add a TODO.